### ISSUE TITLE
show_line_diagnostics() is creating alot of buffers with content of the pum

### ISSUE THREAD

**Issue by lucastrvsn** [neovim/neovim#13429]
- `nvim --version`: NVIM v0.5.0-891-gaec61074a
- language server name/version: All of them
- Operating system/version: Archlinux

<details>
<summary>nvim -c ":checkhealth nvim lspconfig"</summary>

health#nvim#check
========================================================================
## Configuration
  - OK: no issues found

## Performance
  - OK: Build type: Release

## Remote Plugins
  - OK: Up to date

## terminal
  - INFO: key_backspace (kbs) terminfo entry: key_backspace=^H
  - INFO: key_dc (kdch1) terminfo entry: key_dc=\E[3~
  - INFO: $COLORTERM='truecolor'

## tmux
  - OK: escape-time: 1
  - INFO: Checking stuff
  - OK: focus-events: on
  - INFO: $TERM: tmux-256color

health#lspconfig#check
========================================================================
## Checking language server protocol configuration
  - INFO: html: configuration checked.
  - INFO: tsserver: configuration checked.
  - INFO: cssls: configuration checked.
  - INFO: jsonls: configuration checked.
  - INFO: bashls: configuration checked.
  - INFO: vimls: configuration checked.
  - ERROR: rust_analyzer: The given command "rust-analyzer" is not executable.
  - INFO: diagnosticls: configuration checked.
  - INFO: sumneko_lua: configuration checked.

</details>

<details>
<summary>lsp.log</summary>

No error is reported. Empty lsp.log.

</details>

### Steps to reproduce using `nvim -u NORC`

1. Create an file with the following content:

```
call plug#begin()
  Plug 'neovim/nvim-lspconfig'
call plug#end()

lua << EOF
local lsp = require('lspconfig')
lsp.jsonls.setup({})
EOF
```

2. Open neovim with this config file and install the lsp server (doesn't matter which lsp, all of them will show the exactly same problem).

```
nvim -u init_test.vim test.json
```

3. Execute command to show line diagnostics with some error with `:lua vim.lsp.diagnostic.show_line_diagnostics()`

4. List all opened buffers with `:buffers!`

### Actual behaviour

Every time I ask to show line diagnostics, another buffer is opened with the content of the pum of the content created by `show_line_diagnostics()`. This is how I end up finish my work day yesterday (all of the in between buffers exist!!!):

![Captura de tela de 2020-11-30 11-58-03](https://user-images.githubusercontent.com/8826449/100752957-f4975900-33c7-11eb-8947-45e97b55403d.png)

### Expected behaviour

Do not open this amount of buffers and remaining opened every time I want to show diagnostics of one line.

**Note:** I reported this problem [here](https://github.com/neovim/nvim-lspconfig/issues/427) but digging more into this, I discover the problem is not on any lsp, but with the `show_line_diagnostics()`.

**Comment by mfussenegger** [comment]
I think this will be fixed once https://github.com/neovim/neovim/pull/12720 is merged.

**Comment by aktau** [comment]
Why do diagnostics store their contents in buffers? Is that required as input to a popup?

**Comment by lucastrvsn** [comment]
> Why do diagnostics store their contents in buffers? Is that required as input to a popup?

I don't know. Maybe @tjdevries can answer that. He implemented this feature.

Thanks everyone!

**Comment by mfussenegger** [comment]
> Why do diagnostics store their contents in buffers? Is that required as input to a popup?

Yes, floating windows display buffers, so a buffer is needed. See `:help nvim_open_win()`

**Comment by runiq** [comment]
Ephemeral buffers like these should set `bufhidden=wipe`. That's all I did in #12720.

Seeing how #12720 is [back on the drawing board](https://github.com/neovim/neovim/pull/12720#issuecomment-738149567), though, should I factor the `bufhidden=wipe` bit out to a separate PR?
